clouddrift/utility.py

```diff
@@ -0,0 +1,207 @@
+"""
+This module contains various utility functions.
+"""
+
+from clouddrift.ragged import segment, rowsize_to_index
+import numpy as np
+import pandas as pd
+from typing import Optional, Union
+import xarray as xr
+import pandas as pd
+from typing import Optional, Tuple, Union
+from clouddrift.ragged import segment, rowsize_to_index, subset
+
+try:
+    import matplotlib.pyplot as plt
+    import matplotlib.colors as mcolors
+    from matplotlib.collections import LineCollection
+    from matplotlib import cm
+except ImportError:
+    raise ImportError("missing optional dependency 'matplotlib'")
+
+
+def plot_ragged(
+    ax: plt.Axes,
+    longitude: Union[list, np.ndarray, pd.Series, xr.DataArray],
+    latitude: Union[list, np.ndarray, pd.Series, xr.DataArray],
+    rowsize: Union[list, np.ndarray, pd.Series, xr.DataArray],
+    colors: Optional[Union[list, np.ndarray, pd.Series, xr.DataArray]] = None,
+    *args,
+    tolerance: Optional[Union[float, int]] = 180,
+    **kwargs,
+):
+    """Function that wraps matplotlib plot function (plt.plot) and LineCollection
+    (matplotlib.collections) to efficiently plot trajectories from a ragged array dataset.
+
+    Parameters
+    ----------
+    longitude : array-like
+        Longitude sequence. Unidimensional array input.
+    latitude : array-like
+        Latitude sequence. Unidimensional array input.
+    rowsize : list
+        List of integers specifying the number of data points in each row.
+    colors : array-like
+        Colors to use for plotting. If colors is the same shape as longitude and latitude,
+        the trajectories are splitted into segments and each segment is colored according
+        to the corresponding color value. If colors is the same shape as rowsize, the
+        trajectories are uniformly colored according to the corresponding color value.
+    *args : tuple
+        Additional arguments to pass to `plt.plot`.
+    tolerance : float
+        Tolerance gap between data points (in degrees) for segmenting trajectories. For periodic
```

Comment:
Tolerance for longitude?

Reply:
What is the question?

Reply:
"Longitude tolerance gap between..." or "Tolerance gap for longitude between..."